parallel/runner.go

```diff
@@ -237,17 +185,22 @@ func (r *runner) addThread() {
 	nextThreadId := atomic.AddUint32(&r.threadCount, 1) - 1
 	go func(threadId int) {
 		defer r.threadsWaitGroup.Done()
-
-		r.runningThreadsLock.Lock()
-		r.runningThreads++
-		r.runningThreadsLock.Unlock()
+		r.openThreadsLock.Lock()
+		r.openThreads++
+		r.openThreadsLock.Unlock()
 
 		// Keep on taking tasks from the queue.
 		for t := range r.tasks {
-			r.idle.Store(threadId, false)
-
+			// Increase num of threads running tasks
+			atomic.AddUint32(&r.activeThreads, 1)
+			// Mark that some task was executed and the runner started with tasks executing.
+			if !r.started {
+				r.started = true
+			}
 			// Run the task.
 			e := t.run(threadId)
+			// Decrease num of threads running tasks
```

Comment:
// Decrease num of threads running tasks
-->
// Decrease the total of active threads.

Reply:
done